### PR TITLE
Key storage detecting

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

based on #441

- Add JKS and PKCS12 storages detection in --doc mode
- 3 default passwords are used: empty, "changeit" and "changeme" to open a storage

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark - does not impact
- [x] fuzzing

### CODE REVIEW

**Review by csh519 — APPROVED** [review]
Thank you for adding `jks` and `pkcs12` scanner.

LGTM :+1:

File: credsweeper/deep_scanner/jks_scanner.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            self,  #
+            data_provider: DataContentProvider,  #
+            depth: int,  #
+            recursive_limit_size: int) -> List[Candidate]:
+        """Tries to scan JKS to open with standard password"""
+        candidates = []
+        for pw_probe in ["", "changeit", "changeme"]:
+            try:
+                keystore = jks.KeyStore.loads(data_provider.data, pw_probe, try_decrypt_keys=True)
+                if keystore.private_keys or keystore.secret_keys:
```
Comment (kmnls) [inline credsweeper/deep_scanner/jks_scanner.py:27]: This part is not clear.
You check for private keys in the keystore. They are returned in encrypted form (if your password guess is wrong) or decrypted format (if password matches). This is the first part of IF.
Then ELSE section is used for cases when there are no private keys at all. But you marked jks as candidate as well.
It seems there is a trouble out here. 4 cases should be checked independently:
- pkey present, default password, issue.
- pkey present, unknown password, no issue.
- no pkey, default password, no issue (?)
- no pkey, unknown password, no issue.

Comment (babenek) [reply credsweeper/deep_scanner/jks_scanner.py:27]: In any case - a storage with default password should not appear. 
IF-ELSE block just decide which text will be in info for user.

Unknown password will raise an exception and no issue is created.

File: tests/samples/dummy.jks
Comment (kmnls) [inline tests/samples/dummy.jks:1]: Whether it is used? Cannot see traces in the code.

**Review by kmnls — COMMENTED** [review]
See comment.
The code does not match jks source code behavior https://pyjks.readthedocs.io/en/latest/_modules/jks/jks.html#KeyStore.loads

File: credsweeper/deep_scanner/jks_scanner.py (hunk 1 above)
Comment (kmnls) [reply credsweeper/deep_scanner/jks_scanner.py:27]: https://pyjks.readthedocs.io/en/latest/_modules/jks/jks.html#KeyStore.loads
                if try_decrypt_keys:
                    try:
                        entry.decrypt(store_password)
                    except DecryptionFailureException:
                        pass # ok, let user call decrypt() manually
                        
                        
No exception

Comment (babenek) [reply credsweeper/deep_scanner/jks_scanner.py:27]: In common (wrong) case the password of keys is the same like for the storage.
Suppose, if the key has good encryption - else branch will be taken.
Anyway, the scanner finds unencrypted(weak password) storage.

**Review by xDizzix — APPROVED** [review]
+

**Pull request merged**